tools/exp/filter/filter.go

```diff
@@ -0,0 +1,154 @@
+package filter
+
+import (
+	"errors"
+
+	"github.com/aiven/aiven-go-client"
+	"github.com/mitchellh/copystructure"
+	"golang.org/x/exp/maps"
+)
+
+// errUnexpected is the error that is returned when an unexpected error occurs.
+var errUnexpected = errors.New("unexpected filtering error")
+
+// supportedServiceTypes is a pseudo-constant map of supported service types.
+var supportedServiceTypes = func() map[string]struct{} {
+	return map[string]struct{}{
+		"cassandra":                       {},
+		"clickhouse":                      {},
+		"datadog":                         {},
+		"elasticsearch":                   {},
+		"external_aws_cloudwatch_logs":    {},
+		"external_aws_cloudwatch_metrics": {},
+		"external_elasticsearch_logs":     {},
+		"external_google_cloud_logging":   {},
+		"external_kafka":                  {},
+		"external_opensearch_logs":        {},
+		"external_postgresql":             {},
+		"flink":                           {},
+		"grafana":                         {},
+		"influxdb":                        {},
+		"jolokia":                         {},
+		"kafka":                           {},
+		"kafka_connect":                   {},
+		"kafka_mirrormaker":               {},
+		"m3aggregator":                    {},
+		"m3coordinator":                   {},
+		"m3db":                            {},
+		"mysql":                           {},
+		"opensearch":                      {},
+		"pg":                              {},
+		"prometheus":                      {},
+		"redis":                           {},
+		"rsyslog":                         {},
+		"signalfx":                        {},
+	}
+}
+
+// ServiceTypes is a filter that filters out service types that are not supported.
+func ServiceTypes(f map[string]aiven.ServiceType) (map[string]aiven.ServiceType, error) {
+	cf, err := copystructure.Copy(f)
+	if err != nil {
+		return nil, err
+	}
+
+	acf, ok := cf.(map[string]aiven.ServiceType)
+	if !ok {
+		return nil, errUnexpected
+	}
```

Comment:
there are some duplicated codes of copying an object and then casting it to the original type, perhaps we can have a little utility function to make it less duplicated

Reply:
will take a look what could be done about it